### ISSUE TITLE
Randomly fails to show proper autocomplete suggestions for methods/properties despite knowing the type of a variable

### ISSUE THREAD

**Issue by php4fan** [DEVSENSE/phptools-docs#498]
Here:

![image](https://github.com/DEVSENSE/phptools-docs/assets/13969584/129379ea-642d-4283-960b-41166293ee8d)

There's the AI suggestion, but not the known properties and methods for the class.

And after typing a method name partially:

![image](https://github.com/DEVSENSE/phptools-docs/assets/13969584/e9f8a283-f0b8-4f5c-91ec-fffb8dcf1475)

the "abc" suggestions show up, but still no sure suggestions based on the type.

But it knows that `$this->bf` is a `BotFactory`:

![image](https://github.com/DEVSENSE/phptools-docs/assets/13969584/48f653af-e7ee-4dfa-9678-fe00b7617be6)

![image](https://github.com/DEVSENSE/phptools-docs/assets/13969584/ecf50155-f3e3-477d-9d3c-0f4c18afeaa1)

**Comment by php4fan** [comment]
I forgot to mention that this **used to work** as expected with that very same variable, in several other places in my code.

As noticed in another issue, the extension seems to have been updated in the last hours, so maybe this is a regression.

**Comment by Miloslav** [comment]
Thanks.

1.) It appears this is the test case for missing completion:
```php
<?php

class x {

    protected $bf;

    public function __construct(Bar $bar) {
        $this->bf = $bar;
    }

    function test() {
        //$this->bf-> /*Hint on bf is correct BotFactory, but no `foo` is offered.*/       
    }
}

class Bar
{
    public function foo()
    {
    }
}
```
@jakubmisek might know more about this one

2.) AI suggestion in this place is not correct here, right? I assume `getRouter` doesn't exist or what's the context - the code before this.

Thank you

**Comment by php4fan** [comment]
> 2.) AI suggestion in this place is not correct here, right? I assume getRouter doesn't exist or what's the context - the code before this.

`BotFactory::getRouter()` _does_ exist, that's why I expected it to show up as a suggestion (and as the only one when typing part of its name).

The AI suggestion is not _entirely_ correct in that the arguments that it suggests are utter nonsense, but suggesting to call `getRouter()` made perfect sense, indeed it was exactly  the method I wanted to call. Either way, I wasn't concerned about the AI suggestion here.

**Comment by jakubmisek** [comment]
1) fixed and will be available in the next pre-release, thanks!

**Comment by php4fan** [comment]
I keep seeing this issue, although the minimal reproduction example in @Miloslav's comment no longer reproduces.
![image](https://github.com/DEVSENSE/phptools-docs/assets/13969584/2c377482-1894-4d79-8802-f3db7968512c)

```
Published
2018-10-01, 21:22:04
Last released
2024-03-14, 19:58:20
Last updated
2024-03-14, 20:38:08
```

**Comment by php4fan** [comment]
This reproduces:
```
<?php

class x {

    protected $bf;

    public function __construct(Bar $bar) {
        $this->bf = $bar;
    }
}

class y extends x {
    public function test() {
        //$this->bf-> /*Hint on bf is correct Bar, but no `foo` is offered.*/  //<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
    }
}

class Bar
{
    public function foo()
    {
    }
}
```

**Comment by jakubmisek** [comment]
thanks, found the bug, working on it.

**Comment by jakubmisek** [comment]
Fixed in `v1.45.15192`, thanks!
